providers/implementations/rands/seeding/rand_cpu_x86.c

```diff
@@ -12,7 +12,14 @@
 #include "crypto/rand_pool.h"
 #include "prov/seeding.h"
 
-#ifdef OPENSSL_RAND_SEED_RDCPU
+/* Couldn't find how to define arch specific c files, thus necessitating
```

Comment:
It's done in `build.info` looks for the _hw files in providers/implementations/ciphers e.g.

Reply:
That's the thing, we couldn't find an example of an architecture specific C file (we saw asm examples but not C) and the above ifdef exists in other places. Any tip on how to do this in build.info ? It's really not obvious to us...

Reply:
You can put SOURCE directives inside conditionals.  Literally processor-specific conditionals look like
https://github.com/openssl/openssl/blob/6e495312fda0e669d105172c1ac8a8c0bf52da6d/crypto/build.info#L29
but `$config{target}` is much more common than processor-specific checks.

Reply:
@paulidale comment is not quite right there.. The build.info just specify .c files.. The _hw.c files are still just using #ifdef

Reply:
cryptlib.h has this also..

#if defined(__i386)   || defined(__i386__)   || defined(_M_IX86) || \
    defined(__x86_64) || defined(__x86_64__) || \
    defined(_M_AMD64) || defined(_M_X64)
extern unsigned int OPENSSL_ia32cap_P[];
#endif

A #define XX_CAPABLE could be added maybe?

Reply:
fixed by using the $config{target} in `build.info`.